### PR TITLE
DEVELOP-692: Add copy link for branch on github extension

### PR DESCRIPTION
**Pull request by jemmyw**
Add a copy link to github branches in the attribute to make it easier to switch to that branch https://github.com/aha-develop/github/pull/14/files#diff-a8a1f4ed71f557f835fbb57d479877141b86cbb25a9e330abf7cdfd1b383bcc0R8

To make it work better I've also updated the linking so that when a PR is linked it's branch is also linked https://github.com/aha-develop/github/pull/14/files?diff=split&w=1#diff-28e5f92f2fbcf6ab1be7afe60b1c7fe70c742c37b9e4fc1def7762fd4b32d0a7R99 as often the branch name does not match the Aha! reference when the PR does.

https://user-images.githubusercontent.com/8016/119301724-0989e080-bcb7-11eb-882a-baa1060f7b00.mp4

I've also refactored the code a little by pulling the attribute and page components in their own folders. While I was doing that I made the way components are exported the same (named) so there wasn't a confusing mix of default and named exports.

### CODE REVIEW

File: src/components/page/Page.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import React, { useMemo } from "react";
+import BranchTable from "./BranchTable";
+import PrTable from "./PrTable";
+import GithubQuery from "../../lib/query";
+import { useGithubApi } from "../../lib/useGithubApi";
+
+export const Page = ({ repos }) => {
+  const { authed, error, fetchData } = useGithubApi(async () => {});
+
+  const baseQuery = useMemo(
```
Comment (cornu-ammonis) [inline src/components/page/Page.js:10]: so this is intended to never rerun unless the component is remounted?

File: src/components/page/PrTable.js [hunk 2]
```diff
@@ -18,6 +18,13 @@ const refNumMatcher = /([A-Z][A-Z0-9]*-(([E]|[0-9]+)-)?[0-9]+)/;
  * @type {React.FC<RowProps>}
  */
 const PrRow = ({ pr, feature }) => {
+  const showDrawer = (event) => {
```
Comment (cornu-ammonis) [inline src/components/page/PrTable.js:21]: nice

**Review by cornu-ammonis — COMMENTED** [review]
nice refactors! a few questions/comments

File: src/components/Status.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import React from "react";
 import { getPrByUrl, prStatusCommit } from "../lib/github";
 import { useGithubApi } from "../lib/useGithubApi";
 import { usePopperAlerter } from "../lib/usePopperAlerter";
-import ExternalLink from "./ExternalLink";
+import { ExternalLink } from "./ExternalLink";
```
[the remark this answers is not in the document]
Comment (jemmyw) [reply src/components/Status.js:5]: We had some one way, some the other. I just normalized them all to be one way and chose the majority as consensus. Named exports have an advantage imo in that you must refer to their name in the import, making them more greppable. I've run across a couple in aha-app where the component/file had one name and the import referenced it with another.

File: src/components/attribute/Branches.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from "react";
+import { ExternalLink } from "../ExternalLink";
+
+function Branch({ branch }) {
+  const [copied, setCopied] = useState(false);
+
+  const onCopy = () => {
+    navigator.clipboard.writeText(branch.name);
```
[the remark this answers is not in the document]
Comment (jemmyw) [reply src/components/attribute/Branches.js:8]: I don't think it matters too much. I don't want to early return - what if you wanted to copy it to the clipboard again (in less than 1s unlikely, but you never know).

File: src/components/attribute/Branches.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    setTimeout(() => setCopied(false), 1000);
+  };
+
+  return (
+    <aha-flex gap="4px">
+      <span className="type-icon">
+        <aha-icon icon="fa-regular fa-code-branch type-icon" />
+      </span>
+      <ExternalLink href={branch.url}>{branch.name}</ExternalLink>
+      <a
```
[the remark this answers is not in the document]
Comment (jemmyw) [reply src/components/attribute/Branches.js:19]: yeah, it looks a bit crappy as a button. I did try that out. As a link it looks pretty good

File: src/components/page/Page.js (hunk 1 above)
Comment (jemmyw) [reply src/components/page/Page.js:10]: this isn't new code, this is just a component I extracted

File: src/components/attribute/Branches.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        ></aha-icon>
+      </a>
+    </aha-flex>
+  );
+}
+
+export function Branches({ fields }) {
+  if (!fields.branches || fields.branches.length === 0) return null;
+
+  const branches = (fields.branches || []).map((branch, idx) => (
```
[the remark this answers is not in the document]
Comment (jemmyw) [reply src/components/attribute/Branches.js]: seems unlikely but I've updated it anyway

**Review by cornu-ammonis — APPROVED** [review]

File: src/components/Status.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (cornu-ammonis) [reply src/components/Status.js:5]: makes sense, it does bug me when the import differs from the filename for a default export.

**Pull request merged**